lib/wab/impl/data.rb

```diff
@@ -37,27 +38,14 @@ def initialize(value, repair, check=true)
       # same as ['child', 'grandchild'].
       def get(path)
         if path.is_a?(Symbol)
-          node = @root[path] 
+          node = root[path]
         else
           path = path.to_s.split('.') unless path.is_a?(Array)
-          node = @root
-          path.each { |key|
-            if node.is_a?(Hash)
-              node = node[key.to_sym]
-            elsif node.is_a?(Array)
-              i = key.to_i
-              if 0 == i && '0' != key && 0 != key
-                node = nil
-                break
-              end
-              node = node[i]
-            else
-              node = nil
-              break
-            end
-          }
```

Comment:
I'm not seeing the advantage here. The extract_node method is only called from this function which now drops to 7 lines with the build in extract_node. Not much of an improvement as now the reader needs to bounce down to the bottom of the file to see what is going on. I would see it as an improvement if any other method used extract_node but there is none.